### PR TITLE
*: Support sum in new plan

### PR DESCRIPTION
**Pull request by shenli**

### CODE REVIEW

File: ast/functions.go [hunk 1]
```diff
@@ -467,6 +469,32 @@ func (n *AggregateFuncExpr) updateFirstRow() error {
...
+		if err != nil {
+			return errors.Trace(err)
+		}
+		if !d {
+			return nil
+		}
+	}
+	sum := ctx.Value
+	var err error
+	sum, err = types.CalculateSum(sum, value)
```
Comment (qiuyesuifeng) [inline ast/functions.go]: Seems we can directly use `ctx.Value, err = types.CalculateSum(ctx.Value, value)`?

File: util/types/helper.go [hunk 2]
```diff
@@ -67,3 +70,44 @@ func TruncateFloat(f float64, flen int, decimal int) (float64, error) {
...
+	case mysql.Decimal:
+		data = y
+	case nil:
+		data = nil
+	default:
+		data, err = ToFloat64(v)
+	}
+
+	if err != nil {
+		return nil, err
```
Comment (qiuyesuifeng) [inline util/types/helper.go]: errors.Trace(err)

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
PTAL @qiuyesuifeng

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**